monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/data/manager/InMemoryDataManager.java

```diff
@@ -0,0 +1,238 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ *
+ */
+
+package org.wso2.ei.dashboard.core.data.manager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.ei.dashboard.core.exception.DashboardServerException;
+import org.wso2.ei.dashboard.core.rest.delegates.heartbeat.HeartbeatObject;
+import org.wso2.ei.dashboard.core.rest.model.GroupList;
+import org.wso2.ei.dashboard.core.rest.model.NodeList;
+import org.wso2.ei.dashboard.core.rest.model.NodeListInner;
+import java.util.HashMap;
+
+/**
+ * Performs in-memory map operations.
+ */
+public final class InMemoryDataManager implements DataManager {
+
+    private static final Logger logger = LogManager.getLogger(InMemoryDataManager.class);
+    public static HashMap<String, HashMap> heartBeatStore;
+    public static HashMap<String, HashMap> serviceInfoStore;
+
+    public InMemoryDataManager() {
+        heartBeatStore = new HashMap<>();
+        serviceInfoStore = new HashMap<>();
+        logger.debug("heartBeatStore and serviceInfoStore created ");
+    }
+
+    @Override
+    public boolean insertHeartbeat(HeartbeatObject heartbeat, String accessToken) {
+        try {
+            HashMap map = new HashMap<>();
+            map.put("groupId", heartbeat.getGroupId());
+            map.put("nodeId", heartbeat.getNodeId());
+            map.put("interval", heartbeat.getInterval());
+            map.put("mgtUrl", heartbeat.getMgtApiUrl());
+            map.put("timeStamp", heartbeat.getTimestamp());
+            map.put("accessToken", accessToken);
+            String keyString = heartbeat.getGroupId() + heartbeat.getNodeId();
+            heartBeatStore.put(keyString, map);
+            logger.info("Inserted heartbeat to node " + heartbeat.getNodeId() +  " of group " + heartbeat.getGroupId());
+            return heartBeatStore.size() > 0;
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred while inserting heartbeat information.", e);
+        }
+    }
+
+    @Override
+    public boolean insertServerInformation(HeartbeatObject heartbeat, String serverInfo) {
+        try {
+            HashMap map = new HashMap();
+            map.put("groupId", heartbeat.getGroupId());
+            map.put("nodeId", heartbeat.getNodeId());
+            map.put("serviceInfo", serverInfo);
```

Comment:
Seems groupId, nodeId, serviceInfo etc. are used in multiple places. Then shall we define constants for that.

Reply:
fixed